ktor-client/ktor-client-core/common/src/io/ktor/client/features/DefaultResponseValidation.kt

```diff
@@ -25,16 +25,18 @@ public fun HttpClientConfig<*>.addDefaultResponseValidation() {
             val originCall = response.call
             if (statusCode < 300 || originCall.attributes.contains(ValidateMark)) return@validateResponse
 
+            // TODO: change public API in the nearest minor release to make `save` function return SavedHttpCall type
             val exceptionCall = originCall.save().apply {
                 attributes.put(ValidateMark, Unit)
-            }
+            } as? SavedHttpCall ?: throw IllegalStateException("HttpClientCall.save() must return SavedHttpCall")
 
             val exceptionResponse = exceptionCall.response
+            val exceptionResponseText = exceptionCall.responseText
```

Comment:
why we can't just read the content here?

Reply:
Because `readText()` is a suspend function and this file is common code, and `addDefaultResponseValidation` is not marked as suspend

Reply:
Update: you are right, we can, because`validateResponse` takes a `suspend` argument. 
Will fix, thanks.